Mathlib/NumberTheory/Rayleigh.lean

```diff
@@ -0,0 +1,190 @@
+  have h₃ := add_lt_add_of_le_of_lt h₁.1 h₂.1
+  have h₄ := add_lt_add_of_lt_of_le h₁.2 h₂.2
+  simp_rw [← inv_mul_eq_div, ← right_distrib, inv_eq_one_div, hrs.inv_add_inv_conj,
+    one_mul] at h₃ h₄
+  rw [← Int.cast_one] at h₄
+  simp_rw [← Int.cast_add, Int.cast_lt, Int.lt_add_one_iff] at h₃ h₄
+  exact (lt_self_iff_false _).1 (lt_of_le_of_lt' h₄ h₃)
+
+/-- Let `r > 1` and `1/r + 1/s = 1`. Suppose there is an integer `j` where `B_r` and `B'_s` both
+jump over `j` (i.e. an anti-collision). Then this leads to a contradiction. -/
+theorem no_anticollision :
+    ¬∃ (j k m : ℤ), k * r < j ∧ j + 1 ≤ (k + 1) * r ∧ m * s ≤ j ∧ j + 1 < (m + 1) * s := by
+  intro ⟨j, k, m, h₁₁, h₁₂, h₂₁, h₂₂⟩
+  have h₁₁ := (lt_div_iff hrs.pos).2 h₁₁
+  have h₁₂ := (div_le_iff hrs.pos).2 h₁₂
+  have h₂₁ := (le_div_iff hrs.symm.pos).2 h₂₁
+  have h₂₂ := (div_lt_iff hrs.symm.pos).2 h₂₂
+  have h₃ := add_lt_add_of_lt_of_le h₁₁ h₂₁
+  have h₄ := add_lt_add_of_le_of_lt h₁₂ h₂₂
+  simp_rw [← inv_mul_eq_div, ← right_distrib, inv_eq_one_div, hrs.inv_add_inv_conj,
+    one_mul] at h₃ h₄
+  rw [← Int.cast_one, ← add_assoc, add_lt_add_iff_right, add_right_comm] at h₄
+  simp_rw [← Int.cast_add, Int.cast_lt, Int.lt_add_one_iff] at h₃ h₄
+  exact (lt_self_iff_false _).1 (lt_of_le_of_lt' h₄ h₃)
+
+/-- Let `0 < r ∈ ℝ` and `j ∈ ℤ`. Then either `j ∈ B_r` or `B_r` jumps over `j`. -/
+theorem hit_or_miss (h : r > 0) :
+    j ∈ {beattySequence r k | k} ∨ (∃ k : ℤ, k * r < j ∧ j + 1 ≤ (k + 1) * r) := by
+  -- for both cases, the candidate is `k = ⌈(j + 1) / r⌉ - 1`
+  cases lt_or_ge ((⌈(j + 1) / r⌉ - 1) * r) j
+  · refine Or.inr ⟨⌈(j + 1) / r⌉ - 1, by simpa, ?_⟩
+    rw [Int.cast_sub, Int.cast_one, sub_add_cancel, ← div_le_iff h]
+    apply Int.le_ceil
+  · refine Or.inl ⟨⌈(j + 1) / r⌉ - 1, ?_⟩
+    rw [beattySequence, Int.floor_eq_iff]
+    constructor
+    · simpa
+    rw [Int.cast_sub, Int.cast_one, ← lt_div_iff h, sub_lt_iff_lt_add]
+    apply Int.ceil_lt_add_one
+
+/-- Let `0 < r ∈ ℝ` and `j ∈ ℤ`. Then either `j ∈ B'_r` or `B'_r` jumps over `j`. -/
+theorem hit_or_miss' (h : r > 0) :
+    j ∈ {beattySequence' r k | k} ∨ (∃ k : ℤ, k * r ≤ j ∧ j + 1 < (k + 1) * r) := by
+  -- for both cases, the candidate is `k = ⌊(j + 1) / r⌋`
+  cases le_or_gt (⌊(j + 1) / r⌋ * r) j
+  · refine Or.inr ⟨⌊(j + 1) / r⌋, ‹_›, ?_⟩
+    rw [← div_lt_iff h]
+    apply Int.lt_floor_add_one
+  · refine Or.inl ⟨⌊(j + 1) / r⌋, ?_⟩
+    rw [beattySequence', sub_eq_iff_eq_add, Int.ceil_eq_iff, Int.cast_add, Int.cast_one]
+    constructor
+    · rwa [add_sub_cancel]
+    exact sub_nonneg.1 (Int.sub_floor_div_mul_nonneg (j + 1 : ℝ) h)
+
+end Beatty
+
+/-- Generalization of Rayleigh's theorem on Beatty sequences. Let `r` be a real number greater
+than 1, and `1/r + 1/s = 1`. Then the complement of `B_r` is `B'_s`. -/
+theorem rayleigh_compl {r s : ℝ} (hrs : r.IsConjugateExponent s) :
+    {beattySequence r k | k}ᶜ = {beattySequence' s k | k} := by
```

Comment:
```suggestion
theorem compl_beattySequence {r s : ℝ} (hrs : r.IsConjugateExponent s) :
    {beattySequence r k | k}ᶜ = {beattySequence' s k | k} := by
```
and you should also have `compl_beattySequence'`.

Reply:
Done. Also, I added a `'` version for both `compl_beattySequence` and `rayleigh_pos`.